compiler/src/dotty/tools/dotc/transform/patmat/Space.scala

```diff
@@ -354,7 +354,7 @@ class SpaceEngine(implicit ctx: Context) extends SpaceLogic {
     case Bind(_, pat) => project(pat)
     case UnApply(_, _, pats) =>
       if (pat.tpe.classSymbol.is(CaseClass))
-        Kon(pat.tpe.stripAnnots, pats.map(pat => project(pat, roundUp)))
+        Kon(pat.tpe.stripAnnots.dealias, pats.map(pat => project(pat, roundUp)))
```

Comment:
This change fix the exhaustivity check if we remove the `widenForMatchSelector`. But normally, shouldn't `TypeComparer` take care of dealiasing in subtype checking? Ping @odersky @smarter ?